### PR TITLE
Fix for #1135

### PR DESCRIPTION
**Pull request by tobiasqueck**
Fix for #1135 

Local artifacts like controller extensions are not loaded because the requests are forwarded to the backend. This fix intercepts all requests that have a local file representation.

### CODE REVIEW

File: packages/preview-middleware/src/base/flex.ts [hunk 1]
```diff
@@ -3,19 +3,60 @@ import type { ReaderCollection } from '@ui5/fs';
...
+        fragmentPath?: string;
+    };
+}
+
+/**
+ * Map of change type specific correction functions.
+ */
+const moduleNameContentMap: { [key: string]: (change: FlexChange) => string } = {
+    codeExt: (change) => (change.content?.codeRef ?? '').replace('.js', ''),
+    addXML: (change: any) => change.content.fragmentPath
```
Comment (zdravko-georgiev) [inline packages/preview-middleware/src/base/flex.ts:25]: From what I know `change.content` is always defined, even when empty. So we don't really need to check. But if you prefer to keep it, then maybe we can add the check also for the `addXML` use case?

Comment (zdravko-georgiev) [inline packages/preview-middleware/src/base/flex.ts:25]: `addXML: (change) => change.content.fragmentPath ?? ''`?

Comment (tobiasqueck) [reply packages/preview-middleware/src/base/flex.ts]: Yes, https://github.com/SAP/open-ux-tools/pull/1136/commits/d28d2418a8490994d4e6a6d36e4f03497cd1b281

Comment (tobiasqueck) [reply packages/preview-middleware/src/base/flex.ts]: I'd prefer the check since it is read from the filesystem and every 'expert' could modify it.

Comment (zdravko-georgiev) [reply packages/preview-middleware/src/base/flex.ts]: Thx

Comment (zdravko-georgiev) [reply packages/preview-middleware/src/base/flex.ts]: Good point 👍

**Review by zdravko-georgiev — APPROVED** [review]
Code changes look good and are easy to understand
Very good code coverage
Manual testing ✅

**Review by longieirl — APPROVED** [review]
Changes look good.
Havent tested functionality.
Good test coverage.

**Pull request merged**